### PR TITLE
Add test for AutofillGroup api example

### PR DESCRIPTION
**Pull request by IldySilva**
Write Tests for API Examples of `autoffil_group`  as part  of https://github.com/flutter/flutter/issues/130459

This are tests of snippets used in [AutofillGroup class](https://api.flutter.dev/flutter/widgets/AutofillGroup-class.html) 
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Comment by IldySilva** [comment]
@bleroux can you check it,please?

File: examples/api/test/widgets/autofill/autofill_group.0_test.dart [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    final Checkbox checkbox = tester.widget(find.byType(Checkbox));
+    expect(checkbox.value, isTrue);
+
+    // Tap to uncheck the checkbox (to show billing address fields)
+    await tester.tap(find.byType(Checkbox));
+    await tester.pump();
+
+    expect(find.byType(TextField),
+        findsNWidgets(7)); // 5 initial + 2 billing address fields
```
Comment (bleroux) [inline examples/api/test/widgets/autofill/autofill_group.0_test.dart:28]:
```suggestion
    expect(
      find.byType(TextField),
      findsNWidgets(7), // 5 initial + 2 billing address fields
    );  
```
Consider updating the format of the other expect clauses similarly to this proposed change.

**Review by bleroux — COMMENTED** [review]
Thanks for the contribution!

See my comment about the formatting.
Also can you add more checks to verify all TextField's autofillHints value? (Reading the sample code, I see some have `AutofillHints.streetAddressLine2` and also `creditCardNumber`, `creditCardSecurityCode` and `telephoneNumber`.

Because this example is related to the `AutofillGroup` feature, it probably would make sense to check the number of `AutofilGroup` widgets created (similarly to how you checked the number of `TextField`).

**Review by bleroux — COMMENTED** [review]
Minor comments before merging:
- consider adding a check for shippingAddress2, this way all the 7 text fields will be verified.
- small typos, see my specific comments.

**Comment by IldySilva** [comment]
I accidentally remove  the test for shipping address 2, but it's already fixed,  thank you @bleroux

**Review by bleroux — APPROVED** [review]
LGTM! Thanks for your contribution 🙏

**Review by TahaTesser — APPROVED** [review]
LGTM!

**Comment by IldySilva** [comment]
@bleroux   can be merged?🫡

**Pull request merged**